app/src/main/java/com/kickstarter/libs/TrackingClient.kt

```diff
@@ -43,28 +46,44 @@ abstract class TrackingClient(@param:ApplicationContext private val context: Con
 
     final override fun track(eventName: String, additionalProperties: MutableMap<String, Any?>) {
         try {
-            val trackingData = trackingData(eventName, combinedProperties(additionalProperties))
-            val bundle = Bundle()
-            bundle.putString(eventNameKey(), eventName)
-            bundle.putString(eventKey(), trackingData)
-
-            val uniqueJobName = backgroundServiceClass().simpleName + System.currentTimeMillis()
-            dispatchJob(this.context, backgroundServiceClass(), uniqueJobName, bundle)
+            queueEvent(eventName, additionalProperties)
+        } catch (e: JSONException) {
             if (this.build.isDebug) {
-                Log.d(tag(), "Queued event:$trackingData")
+                Timber.e("Failed to encode ${type().tag} event: $eventName")
             }
-        } catch (e: JSONException) {
+            Fabric.getLogger().e(TrackingClient::class.java.simpleName, "Failed to encode ${type().tag} event: $eventName")
+        }
+    }
+
+    private fun queueEvent(eventName: String, additionalProperties: MutableMap<String, Any?>) {
+        val eventData = trackingData(eventName, combinedProperties(additionalProperties))
+        val data = workDataOf(IntentKey.TRACKING_CLIENT_TYPE_TAG to type().tag,
+                IntentKey.EVENT_NAME to eventName,
+                IntentKey.EVENT_DATA to eventData)
+
+        var requestBuilder: OneTimeWorkRequest.Builder? = null
+        if (type() == Type.KOALA) {
+            requestBuilder = OneTimeWorkRequestBuilder<KoalaWorker>()
+        } else if (type() == Type.LAKE) {
+            requestBuilder = OneTimeWorkRequestBuilder<LakeWorker>()
+        }
+
+        requestBuilder?.let {
+            val request = it
+                    .setInputData(data)
+                    .setBackoffCriteria(BackoffPolicy.EXPONENTIAL, 30, TimeUnit.SECONDS)
```

Comment:
Can we reuse the `BACKOFF_DELAY` from the DeviceRegistrar?

Reply:
I'd rather give it its own, and I think I'll change it to `10` since tracking events will be happening more frequently.

Reply:
But I think it makes the code less readable
```
            val backOffDelay = 10
            val request = it
                    .setInputData(data)
                    .setBackoffCriteria(BackoffPolicy.EXPONENTIAL, backOffDelay, TimeUnit.SECONDS)
                    .setConstraints(baseConstraints)
                    .build()
```

Reply:
If its only used in one place, then we don't need the constant. I didn't know if those values were the same. Changing it to 10 makes sense